### PR TITLE
added VICReg losses

### PR DESCRIPTION
**Pull request by drscotthawley**
This adds a few very simple functions that implement VICReg. 

If the style of writing these are not to your liking, let me know how I can be more compliant with the style 

-- e.g. I didn't find it necessary to create nn.Module Classes for these since they are so simple, but I could so.

### CODE REVIEW

File: oobleck/losses.py [hunk 1]
```diff
@@ -170,3 +170,26 @@ def forward(self, inputs: TensorDict) -> TensorDict:
...
         return inputs
+
+
+
+# VICReg, https://arxiv.org/abs/2105.04906
+# Typically there will already be some loss in use to enforce some similarity and draw points together.
+# These two losses offer a sort of "repulsion" regularization, across the batch dimension, to keep the
+# space from collapse.  Intended to be connected to the latents in the middle of the autoencoder..
+
+def vicreg_var_loss(z, gamma=1, eps=1e-4):
```
Comment (zqevans) [inline oobleck/losses.py:177]: These need to made into instantiable "loss modules" like the ones above. Also please add a test for them.

**Comment by drscotthawley** [comment]
@zqevans take a look at the new version; not sure I wrote the module class correctly.

File: oobleck/losses.py [hunk 2]
```diff
@@ -192,4 +188,31 @@ def vicreg_cov_loss(z):
...
+
+    def __init__(self, gamma: float = 1., eps: float = 1.0e-4, weight: float = 1.) -> None:
+        super().__init__()
+        self.gamma, self.eps = gamma, eps
+        self.weight = weight
+
+    def forward(self, inputs: TensorDict) -> TensorDict:
+        z = inputs['latent']
+        # writing with self.... "hooks" to enable extraction of  individual losses for logging
+        self.var_loss = vicreg_var_loss(inputs, gamma=self.gamma, eps=self.eps)
```
Comment (zqevans) [inline oobleck/losses.py:208]: Shouldn't store values in instance variables like this, doesn't make sense semantically. 

If you want to return the individual losses, you can do so in the `inputs.update()` call on line 217.

File: oobleck/losses.py [hunk 3]
```diff
@@ -192,4 +188,31 @@ def vicreg_cov_loss(z):
...
+        self.cov_loss = vicreg_cov_loss(inputs)
+        loss = self.var_loss + self.cov_loss
+        inputs = accumulate_value(
+            inputs,
+            {
+                "vicreg_loss": self.weight * loss
```
Comment (zqevans) [inline oobleck/losses.py:214]: Should be "generator_loss". This line adds the vic_reg loss to the overall accumulated loss while the `inputs.update()` line below is used to return the individual named losses.
